### ISSUE TITLE
Compatibility Platform with higher Target Framework	

### ISSUE THREAD

**Issue by SHuygh** [Samsung/TizenFX#1604]
Is it possible to deploy to a lower Platform i.e. Tizen 4.0.0 when targeting Tizen50.

We are having some crashes using Tizen.Net.Api4 which are fixed in Tizen.Net.Api5
https://github.com/Samsung/TizenFX/blob/API5/src/ElmSharp/ElmSharp/EvasObject.cs

 "exception" : "InvalidOperationException Collection was modified; enumeration operation may not execute.", 
"stackTrace" : "
at System.Collections.Generic.HashSet`1.Enumerator.MoveNext()
at ElmSharp.EvasObject.DisposeEvent() at ElmSharp.EvasObject.Unrealize()
at Xamarin.Forms.Platform.Tizen.VisualElementRenderer`1.Dispose(Boolean disposing)
at Xamarin.Forms.Platform.Tizen.LayoutRenderer.Dispose(Boolean disposing)
at Xamarin.Forms.Platform.Tizen.VisualElementRenderer`1.Dispose()
at Xamarin.Forms.Platform.Tizen.VisualElementRenderer`1.Dispose(Boolean disposing)
at Tizen.Wearable.CircularUI.Forms.Renderer.CirclePageRenderer.<>n__0(Boolean disposing) 
at....

This crash is currently the biggest issue in our application thus are searching for a way to fix it.

When we target Tizen50 we are getting a crash at startup stating:
System.IO.FileLoadException: Could not load file or assembly 'System.Runtime.Extensions, Version=4.2.1.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'. The located assembly's manifest definition does not match the assembly reference. (Exception from HRESULT: 0x80131040)

Do you have any suggestions?

**Comment by WonyoungChoi** [comment]
@SHuygh  Unfortunately, It is not possible. Tizen 4.0 platform supports only API Level 4 of TizenFX.

**Comment by SHuygh** [comment]
Thank you for the information. Created a PR https://github.com/Samsung/TizenFX/pull/1696 to fix the issue for API4

**Comment by myroot** [comment]
Could you tell me a crash reproduce scenario?
Actually, On API4, _eventStore was never updated on DisposeEvent(). because does not anything on Dispose of SmartEvent/EvasObjectEvent

https://github.com/Samsung/TizenFX/blob/43898f684801b3423daed7f1478e512702aa7849/src/ElmSharp/ElmSharp/SmartEvent.cs#L174
https://github.com/Samsung/TizenFX/blob/43898f684801b3423daed7f1478e512702aa7849/src/ElmSharp/ElmSharp/EvasObjectEvent.cs#L354



After, in API5 I made this patch, I need to update DisposeEvent method, because Dispose of SmartEvent/EvasObjectEvent is modify _eventStore list, so I make snapshot of _eventStore. 
https://github.com/Samsung/TizenFX/commit/f4f56d0ac67df76525715a1eaf9fedcb720e9297#diff-fbdd22d74aad71adfcb35cf8b2f07e35L354

**Comment by SHuygh** [comment]
I wish I had a reproducible scenario. I have never been able to have this crash in our development environment, however the users have it in the production environment.
As it is an application that is used in a professional environment we have about 700 users at the moment which use the application for 6-8 hours a day and have had the crash about 200 times in the last 30 days.
Thus it does not occur frequently, but too frequent as the application has a security function and limits our users in their work.

The issue will be caused when either MakeInvalidateEvent() or AddToEventLifeTracker(IInvalidatable item) are called on a different thread than the DisposeEvent()

The issue is within the 3rd layer of abstraction for us (CircularUI -> Xamarin.Forms -> TizenFx), therefor my insight here is not extensive enough to have a better explanation.

We often add and remove screens to an IndexPage which might be related.

**Comment by myroot** [comment]
You should not acess UI API on others thread.
UI API must be used only main thread.
Please check your code which code was use worker therad

**Comment by SHuygh** [comment]
I have checked our code base, and we might have had issues with the navigation not occurring on the main thread.
I currently reworked our service responsible for navigation to ensure that the navigation always is dispatched to the main thread if required.
As we do not have a reproducible scenario it was hard to test if the issue is fixed or not. I tried explicitly dispatching our navigation to the threadpool, which resulted in some cases (depending on race conditions) in .NET exceptions, app crashes and UI issues which have been reported before by our users.
Currently we are starting up migrating the build containing the fix to our production environment, after which it will become clear if the current mentioned issue is indeed fixed.
